jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/PathMappings.java

```diff
@@ -201,6 +201,8 @@ public static PathSpec asPathSpec(String pathSpecString)
     {
         if ((pathSpecString == null) || (pathSpecString.length() < 1))
         {
+            if (pathSpecString != null)
```

Comment:
The conditions are quite convoluted when written this way. Can't they be rewritten in a clearer way like the following?
```java
if (pathSpecString == null)
    throw new RuntimeException("Path Spec String must start with '^', '/', or '*.': got [" + pathSpecString + "]");
if ("".equals(pathSpecString))
    return new ServletPathSpec("");
return pathSpecString.charAt(0) == '^' ? new RegexPathSpec(pathSpecString) : new ServletPathSpec(pathSpecString);
```

Reply:
evolved solutions are mostly ugly!  will do!